go-controller/pkg/ovn/egressip.go

```diff
@@ -1945,6 +1934,82 @@ func (e *egressIPZoneController) deleteEgressIPStatusSetup(name string, status e
+	// it doesn't match with nexthops derived from eIP.Status.
+	policyPred := func(item *nbdb.LogicalRouterPolicy) bool {
+		return item.Priority == types.EgressIPReroutePriority && item.ExternalIDs["name"] == name
+	}
+	eIP, err := e.watchFactory.GetEgressIP(name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return ops, fmt.Errorf("error retrieving EgressIP %s object for updating logical router policy nexthops, err: %w", name, err)
+	} else if err != nil && apierrors.IsNotFound(err) {
+		// EgressIP object is not found, so delete LRP associated with it.
+		ops, err = libovsdbops.DeleteLogicalRouterPolicyWithPredicateOps(e.nbClient, ops, e.GetNetworkScopedClusterRouterName(), policyPred)
+		if err != nil {
+			return ops, fmt.Errorf("error creating ops to remove logical router policy for EgressIP %s from router %s: %v",
+				name, e.GetNetworkScopedClusterRouterName(), err)
+		}
+	} else {
+		validNextHopIPs := make(sets.Set[string])
+		for _, validStatus := range eIP.Status.Items {
+			nextHopIP, err := e.attemptToGetNextHopIP(name, validStatus)
+			if err != nil {
+				return ops, fmt.Errorf("failed to delete EgressIP %s (%s) because unable to determine next hop: %v",
+					name, validStatus.EgressIP, err)
+			}
+			validNextHopIPs.Insert(nextHopIP)
+		}
+
+		reRoutePolicies, err := libovsdbops.FindLogicalRouterPoliciesWithPredicate(e.nbClient, policyPred)
+		if err != nil {
+			return ops, fmt.Errorf("error finding logical router policy for EgressIP %s: %v", name, err)
+		}
+		for _, policy := range reRoutePolicies {
+			for _, nextHop := range policy.Nexthops {
+				if validNextHopIPs.Has(nextHop) {
+					continue
+				}
+				ops, err = libovsdbops.DeleteNextHopsFromLogicalRouterPolicyOps(e.nbClient, ops, e.GetNetworkScopedClusterRouterName(), []*nbdb.LogicalRouterPolicy{policy}, nextHop)
+				if err != nil {
+					return ops, fmt.Errorf("error creating ops to remove stale next hop IP %s from logical router policy for EgressIP %s from router %s: %v",
+						nextHop, name, e.GetNetworkScopedClusterRouterName(), err)
+				}
+			}
+		}
+	}
+	return ops, nil
+}
+
+// attemptToGetNextHopIP this function attempts to retrieve nexthops associated with logical router policy for the given
+// EgressIP's status. It ensures the following conditions are met.
+// 1) When node is not found, then it must ruturn empty nexthop without an error.
+// 2) When EgressIP belongs to OVN network and node is local, then it must return node's gateway router IP address.
+// 3) When EgressIP belongs to non OVN network and node is local, then it must return node's management port IP address.
+// 4) When EgressIP belongs to remote node in interconnect zone, then it return node's transit switch IP address.
+func (e *egressIPZoneController) attemptToGetNextHopIP(name string, status egressipv1.EgressIPStatusItem) (string, error) {
+	isLocalZoneEgressNode, _ := e.nodeZoneState.Load(status.Node)
+	eNode, err := e.watchFactory.GetNode(status.Node)
+	if err != nil {
```

Comment:
dont you want to check here specifically that the node doesn't exist, instead of some other error? 

!apierrors.IsNotFound(err)